### PR TITLE
chore: upgrade golang to 1.16

### PR DESCRIPTION
**Pull request by Demonsthere**
- Update golang to 1.16
- Import CVE checking pipeline

## Related Issue or Design Document
- Fixes https://github.com/ory/hydra-maester/issues/102

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: .github/workflows/cve-scan.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Build images
+        shell: bash
+        run: |
+          make docker-build-notest
+      - name: Scan image
+        uses: anchore/scan-action@v3
+        with:
+          image: controller:latest
+          fail-build: true
+          severity-cutoff: high
```
Comment (Demonsthere) [inline .github/workflows/cve-scan.yaml:44]:
```suggestion
          # severity-cutoff: high
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -42,8 +44,8 @@ jobs:
           name: Update golang
           command: |
             sudo rm -rf /usr/local/go/
-            curl -LO https://dl.google.com/go/go1.15.11.linux-amd64.tar.gz
-            sudo tar -C /usr/local -xzf go1.15.11.linux-amd64.tar.gz
+            curl -LO https://go.dev/dl/go1.16.10.linux-amd64.tar.gz
+            sudo tar -C /usr/local -xzf go1.16.10.linux-amd64.tar.gz
```
Comment (zepatrik) [inline .circleci/config.yml:48]: There is also the setup go action: https://github.com/actions/setup-go

File: Makefile [hunk 3]
```diff
@@ -1,94 +1,137 @@
...
 vet:
 	go vet ./...
 
 # Generate code
+.PHONY: generate
 generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths=./api/...
 
 # Build the docker image
+.PHONY: docker-build-notest
```
Comment (zepatrik) [inline Makefile:107]: When building the docker image, I have to previously build the manager binary locally right? So the docker image has no go installed?

File: .circleci/config.yml (hunk 2 above)
Comment (Demonsthere) [reply .circleci/config.yml:48]: That is circleCi ;)

File: Makefile (hunk 3 above)
Comment (Demonsthere) [reply Makefile:107]: Yeah, the base images here are distroless, so we need to build the image outside and copy it. We might want to migrate to a multistage dockerfile tho

**Review by zepatrik — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (zepatrik) [reply .circleci/config.yml:48]: oh damn :joy:

**Pull request merged**